topostats/unet_masking.py

```diff
@@ -222,63 +219,94 @@ def make_bounding_box_square(
     crop_min_col : int
         The minimum column index of the crop.
     crop_max_row : int
-        The maximum row index of the crop.
+        The maximum EXCLUSIVE row index of the crop.
     crop_max_col : int
-        The maximum column index of the crop.
+        The maximum EXCLUSIVE column index of the crop.
     image_shape : tuple[int, int]
         The shape of the image.
 
     Returns
     -------
     tuple[int, int, int, int]
         The new crop indices.
+
+    Notes
+    -----
+    The crop indices are inclusive on the minimum side and exclusive on the maximum side.
+    So an object like this:
+    0 0 0 0 0
+    0 1 1 1 0
+    0 1 1 1 0
+    0 1 1 1 0
+    0 0 0 0 0
+    With no padding, would have a bounding box of (1, 1, 4, 4) and not (1, 1, 3, 3).
+    Hence the maximum index for the bbox is allowed to exceed the maximum index of the image by 1.
```

Comment:
I noticed in one of the tests though that a 10x10 image had a `bbox` where the max values were `12` which is more than 1, but everything works (by the looks of it!).

Reply:
Fixed I think, though there are currently no checks on the fly to ensure that the bbox is correct since the only time it would be wrong is if we manually set it to be wrong (like when constructing tests!)

Could add an getter-setter check later on if this becomes an issue though 😄 (personally love getter-setter sanity checking)